app/controllers/users.py

```diff
@@ -1064,16 +1065,29 @@ def get(self):
             returned_users = self.computed_results[date_range]
 
         else:
-            query = User.query.filter(
-                cast(User.last_seen, Date) <= start_date,
-                cast(User.last_seen, Date) >= end_date,
-                User.verified == True
-            )
-            if created_date is not None:
-                query = query.filter(
-                    cast(User.date_created, Date) <= today,
-                    cast(User.date_created, Date) >= created_date,
+            if (keywords) :
```

Comment:
Instead of repeating the same lines of code, try something like this
```suggestion   
query = User.query.filter(
    cast(User.last_seen, Date) <= start_date,
    cast(User.last_seen, Date) >= end_date,
    User.verified == True
   )

if keywords:
    keyword_filter = (User.name.ilike('%' + keywords + '%') | User.email.ilike('%' + keywords + '%'))
    query = query.filter(keyword_filter)

if created_date:
    date_created_filter = (
        cast(User.date_created, Date) <= today,
        cast(User.date_created, Date) >= created_date
    )
    query = query.filter(date_created_filter)

```

Reply:
Ok man thanks for that . It is cleaner...